packages/remix-react/browser.tsx

```diff
@@ -1,7 +1,5 @@
-// TODO: We eventually might not want to import anything directly from `history`
-// and leverage `react-router` here instead
-import type { BrowserHistory, Update } from "history";
-import { createBrowserHistory } from "history";
+import type { BrowserHistory, Action, Location } from "@remix-run/router";
+import { createBrowserHistory } from "@remix-run/router";
```

Comment:
Normally, `@remix-run/react` should be importing from `react-router-dom`.  These are the only imports from `@remix-run/router` since history stuff isn't re-exported through `react-router-dom` since users aren't supposed to do manual history management in 6.4.